commands/polls/createPoll.js

```diff
@@ -0,0 +1,108 @@
+const logger = require('../../utils/logger');
+
+module.exports = {
+    data: {
+        name: 'createPoll',
+        type: 'text'
+    },
+    
+    async execute(message) {
+        const guildId = message.guildId;
+        const userName = message.member.user.username;
+        logger.log(`!createPoll user: ${userName} | guildId: ${guildId}`);
+        let [doPoll, outputStr] = parseArgs(message.content);
+        if (doPoll) {
+            await message.author.send('Here is your poll:\n' + outputStr + '\nDoes this look good?');
+            await message.author.dmChannel.awaitMessages({
+                max: 1,
+                time: 180000,
+                errors: ['time']
+            }).then((response) => {
+                if (response.first().content.toUpperCase().startsWith('Y')) {
+                    message.author.dmChannel.send('Your poll has been posted.');
+                    logger.log(`${userName} has created a public poll. | guildId: ${guildId}\n`);  
+                } else {
+                    message.author.dmChannel.send('Your poll was not posted');
+                    outputStr = 'Poll not posted.';
+                    return;
+                }
+            }).catch(() => {
+                message.author.send('Something went wrong! Poll not posted.');
+                return;
+            }); 
+        } 
+        if (outputStr != '') {
+            await message.channel.send(`${outputStr}`);
+        }
+    }
+
+};
+
+const parseArgs = (args) => {
```

Comment:
this code is alright (aside from other comments) for now but we should work on a generic args parsing solution so we can avoid all of this logic in the future. It seems very brittle and unflexible

Reply:
yeah dude smart users